kong/runloop/balancer.lua

```diff
@@ -134,7 +134,12 @@ do
       -- split `target` field into `name` and `port`
       local port
       target.name, port = string.match(target.target, "^(.-):(%d+)$")
-      target.port = tonumber(port)
+      if target.name and port then
+        target.port = tonumber(port)
```

Comment:
what if `port` is greater than `65535`?

Reply:
isn't that caught by schema validations?

Reply:
Yes, schema uses `kong.tools.utils.normalize_ip` to validate IP and port.